src/Containers/MinimalContainers/ConstantSizeMatrix.hpp

```diff
@@ -160,13 +160,26 @@ class ConstantSizeMatrix
 
   void resize(size_t m, size_t n)
   {
-    if (n > n_max_ || m > m_max_)
+    if (n * m > n_max_ * m_max_)
     {
       std::ostringstream error;
       error << "You cannot resize a constant size matrix beyond its initial max dimensions ( " << m << "," << n << " > "
             << m_max_ << "," << n_max_ << " )\n";
       throw std::domain_error(error.str());
     }
+
+    size_t capacity = n_max_ * m_max_;
```

Comment:
calculate capacity before line 163

Reply:
Done.  I also changed the name of the variable from `capacity` to `max_capacity` to avoid confusion with the `capacity()` function.